### ISSUE TITLE
Graph.remove_node raises NetworkXError while networkx.relabel_nodes raises KeyError

### ISSUE THREAD

**Issue by blthayer** [networkx/networkx#4465]
For starters, thank you for this great package! On to the issue:

As the title says, `Graph.remove_node` raises a `NetworkXError` if the node is not present, while `networkx.relabel_nodes` raises a `KeyError` if a node is present in the `mapping` but not the graph (`G`). This feels inconsistent.

Additionally, the [documentation for remove_node](https://networkx.org/documentation/stable/reference/classes/generated/networkx.Graph.remove_node.html) explicitly mentions a `NetworkXError` will be raised, while the [documentation for relabel_nodes](https://networkx.org/documentation/stable/reference/generated/networkx.relabel.relabel_nodes.html) does not specify anything related to exceptions (except to say a circular mapping raises an "exception" for in-place relabeling).

Finally, it's worth noting that the [Exceptions documentation](https://networkx.org/documentation/stable/reference/exceptions.html) lists a `NodeNotFound` exception that neither `remove_node` nor `relabel_nodes` raises.

**Comment by dschult** [comment]
Can you give a short example of `relabel_nodes` raising a `KeyError`?  
Because it is not supposed to matter whether the mapping includes fewer or more nodes than are contained in G.
For example:
```python
G=nx.path_graph(9)
H=nx.relabel_nodes(G, {1:2, 2:1, 11:13})
```
works for me.  With copy=False I get a `NetworkXUnfeasible` error.

Also, I don't think a KeyError for a missing key in the dict mapping and a NetworkXError for a node missing from G is inconsistent. A dict can have any key -- not just nodes. While a Graph uses nodes as its keys.

The `NodeNotFound` was an experimental attempt to move toward having many error types within the NetworkX package. It became too much to maintain with little benefit since the error message is present either way.  But we should probably figure out what to do with that.  Care to write a NXEP for how best to organize our error types?

**Comment by dschult** [comment]
Ahh...  The corner case where the KeyError is raised is:
```python
G=nx.path_graph(9)
nx.relabel_nodes(G, {1:"A", 2:"B", 11:13}, copy=False)
```
The other trouble with this corner case is that the mapping has been partially applied by the time this error occurs. So it leaves the relabel in a bad state that is not easy to predict.

I'd like to figure out if it would work to `continue` rather than raise any error.
Otherwise, I think we should make sure the whole relabel will work before we start doing it.

**Comment by dschult** [comment]
It will indeed work to have the code ignore extra keys in the mapping that are not nodes in the graph.
Ignoring that entry is already done when copy=True and doesn't cause any issues when copy=False.

Making that change would be a change to the API and could mess up a user who relies on `nx.relabel_nodes(G, mapping, copy=False)` raising a KeyError when a key of mapping is not a node in G.  Would anybody do that?

@jarrodmillman, would this change need to be a deprecation since the code wouldn't raise an error anymore, or can we treat it as adding a feature where we don't need to deprecate?  To be clear, it isn't completely settled that ignoring keys that aren't in G is the right thing to do. But I think it is...

**Comment by jarrodmillman** [comment]
It seems reasonable to just add a comment to the release notes.  I don't think this requires a deprecation.

**Comment by blthayer** [comment]
@dschult @jarrodmillman Thank you both for your quick responses to this issue.

Unfortunately I don't have the bandwidth (nor the NetworkX expertise :) to write out an NXEP at the moment, but did want to bring this issue to your attention.

To summarize my view succinctly: The [EAFP](https://docs.python.org/3/glossary.html) coding style common in Python means many developers (such as myself) will write code that attempts to "do something" to a graph in NetworkX and "not care" if it succeeds or not. For example:

```python

try:
    my_graph.remove_node(2)
except NetworkXError:
    # I'm not worried about this error, though I may inspect the message if I am.
    # This is just a standard part of my flow control, subscribing to EAFP style
    continue
```

However, leveraging the EAFP style becomes a bit more difficult (or at least makes user code less modular/reusable) without a unified and defined exception hierarchy. If I have to handle a `KeyError` in one place when a node isn't present but a (relatively generic) `NetworkXError` elsewhere (for the same root cause - missing a node).

I appreciate the rapid pull request from Mr. Schult and the discussion here. Thanks again for NetworkX - it's a wonderful package!

**Comment by rossbar** [comment]
Closed by #4466
